main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -813,80 +805,9 @@ object CompletionProvider {
-    */
-  private def getImportMethodCompletions()(implicit context: CompletionContext): Iterable[CompletionItem] = {
-    val instance = raw"\s*import\s+(.*)".r
-    val static = raw"\s*import\s+static\s+(.*)".r
-    context.prefix match {
-      // We match on static first because import static ... would also match on instance regex.
-      case static(clazz) => methodsCompletion(clazz, true)
-      case instance(clazz) => methodsCompletion(clazz, false)
-      case _ => Nil
-    }
-  }
-
-  /**
-    * Convert methods of a class into completionitems
-    */
-  private def methodsCompletion(clazz: String, isStatic: Boolean)(implicit context: CompletionContext): Iterable[CompletionItem] = {
-    classFromDotSeperatedString(clazz) match {
-      case Some((clazzObject, clazz)) => clazzObject.getMethods()
-        // Filter if the method is static or not.
-        .filter((method) => java.lang.reflect.Modifier.isStatic(method.getModifiers()) == isStatic)
-        .map((method) => executableCompletion(method, clazz, None))
-      case None => Nil
-    }
-  }
-
-  /**
-    * returns a completion from a java executable (method/constructor) instace.
-    * clazz is the clazz in string form used for the completion.
-    * aliasSuggestion is used to suggest a alias for the function if applicable.
-    */
-  private def executableCompletion(exec: Executable, clazz: String, aliasSuggestion: Option[String])(implicit context: CompletionContext): CompletionItem = {
-    val typesString = exec.getParameters().map(param => convertJavaClassToFlixType(param.getType())).mkString("(", ", ", ")")
-    val finalAliasSuggestion = aliasSuggestion match {
-      case Some(aliasSuggestion) => s" as $${0:$aliasSuggestion}"
-      case None => ""
-    }
-    // Get the name of the function if it is not a constructor.
-    val name = if (exec.isInstanceOf[Constructor[_ <: Object]]) "" else s".${exec.getName}"
-    // So for constructors we do not have a return type method but we know it is the declaring class.
-    val returnType = if (exec.isInstanceOf[Method]) exec.asInstanceOf[Method].getReturnType() else exec.getDeclaringClass()
-    val label = s"$clazz$name$typesString"
-    val replace = s"$clazz$name$typesString: ${convertJavaClassToFlixType(returnType)} \\ IO$finalAliasSuggestion;"
-    CompletionItem(
-      label = label,
-      // Prioritise executables with fewer parameters.
-      sortText = Priority.high(s"${exec.getParameterCount()}$label"),
-      textEdit = TextEdit(context.range, replace),
-      documentation = None,
-      insertTextFormat = InsertTextFormat.Snippet,
-      kind = CompletionItemKind.Method)
+  private def getImportCompletions()(implicit context: CompletionContext, index: Index, root: Option[TypedAst.Root], delta: DeltaContext): Iterable[CompletionItem] = {
+    if (root.isEmpty) Nil else (ImportNewCompleter.getCompletions ++ ImportMethodCompleter.getCompletions map (comp => comp.toCompletionItem)) ++
```

Comment:
Instead of calling ++ and then toCompletionItem multiple times, you can first call ++ and then at the very end map and call toCompletionItem.

Reply:
Also, this is just a online method, so why not inline it where it is used?